src/tests/test_integration.py

```diff
@@ -954,3 +964,15 @@ def foo():
     _, _, code = env.invoke()
 
     assert code == 0
+
+
+def test_syntax_error_multiple_files(env):
```

Comment:
Well if a single syntax error prevented further checking, then wouldn't it hold that we would never see the second one produce a syntax error either? I don't think a third file is needed here.